signac/contrib/project.py

```diff
@@ -1526,11 +1518,45 @@ def init_project(cls, name, root=None, workspace=None, make_dir=True):
             configuration.
 
         """
+        # TODO: Remove both the `if args` and `if kwargs` blocks in version 3.0
+        # when we remove backwards compatibility for project name APIs.
+        if args:
+            num_args = len(args)
+            if num_args == 1:
+                warnings.warn(
+                    "Project names were removed in signac 2.0. If you intended to call "
+                    "`init_project` with a root directory, please provided it as a keyword "
+                    f"argument: `init_project(root={args[0]})`. If your project name "
+                    "contains important information, consider storing it in the project document "
+                    "instead.",
+                    FutureWarning,
+                )
+            else:
+                # Match the usual error from misusing keyword-only args.
+                raise TypeError(
+                    f"init_project() takes 0 positional arguments but {num_args} "
+                    f"were given"
+                )
+        if kwargs:
+            name = kwargs.pop("name", None)
+            if kwargs:
+                # Match the usual error from extra keyword args.
+                raise TypeError(
+                    f"init_project() got an unexpected keyword argument '{next(iter(kwargs))}'"
+                )
+            if name is not None:
+                warnings.warn(
+                    "Project names were removed in signac 2.0. If your project name contains "
+                    "important information, consider storing it in the project document instead.",
+                    FutureWarning,
+                )
+
         if root is None:
             root = os.getcwd()
         try:
             project = cls.get_project(root=root, search=False)
         except LookupError:
+            name = "project"
```

Comment:
It seems like this `except` block might be doing things that we no longer want to do? e.g. `config["project"] = name` and `assert project.id == str(name)`?

Reply:
See https://github.com/glotzerlab/signac/pull/677#discussion_r805193191. For now, project names are always "project" upon initialization, but they do still exist and the id is well-defined, so I left those checks intact. I will remove all of that logic in a separate PR removing project names from the config.